src/utils/solana/getPrivateKey.ts

```diff
@@ -0,0 +1,14 @@
+import { etc, getPublicKey } from '@noble/ed25519'
+import { sha512 } from '@noble/hashes/sha512'
+import bs58 from 'bs58'
+
+// pk is in hex format with 64 length from key_helper
+function getPrivateKey(pk: string) {
```

Comment:
You don't necessarily have to use the synchronous function, if it's feasible to make this asynchronous, you can just use `getPublicKeyAsync` and avoid this whole thing (it'll also use WebCrypto that way IIRC).

Reply:
Fixed here https://github.com/arcana-network/wallet-ui/pull/535/commits/c1c34d09b26fc0a6774942137b4a20ff5fe9cbe4